tests/e2e/contracts/probes/observability_probe.py

```
@@ -234,11 +234,23 @@ async def _check_stream_exists(c: httpx.AsyncClient) -> bool:
             return await _check_stream_exists(c)
 
     async def _get_trace(self, trace_id: str) -> dict[str, Any]:
-        payload = await self._search_openobserve(
-            f"SELECT * FROM traces WHERE trace_id='{trace_id}'",
-            signal_type="traces",
-            limit=200,
-        )
+        end = datetime.now(timezone.utc)
+        start = end - timedelta(minutes=30)
+        params = {
+            "query": f"SELECT * FROM traces WHERE trace_id='{trace_id}'",
```

Ruff (0.15.11)

[error] Possible SQL injection vector through string-based query construction

(S608)